app/controllers/application_controller.rb

```diff
@@ -19,4 +20,11 @@ def redirect_to_setup
   def setup_done?
     !Minion.assigned_role.count.zero?
   end
+
+  # TODO: (mssola) remove once we have a reverse-proxy setup in place.
+  def redirect_to_secure
+    return if params[:action] == "autoyast" || params[:controller] == "health"
```

Comment:
I think you can modify the `before_action` so you can do: `before_action :redirect_to_secure, unless: [:autoyast, :health]`